src/MICore/LaunchOptions.cs

```diff
@@ -1849,6 +1909,39 @@ public static MIMode ConvertMIModeAttribute(Xml.LaunchOptions.MIMode source)
             Debug.Assert((uint)MIMode.Clrdbg == (uint)Xml.LaunchOptions.MIMode.clrdbg, "Enum values don't line up!");
             return (MIMode)source;
         }
+
+        protected static string ParseArguments(IEnumerable<string> arguments, bool alwaysQuote = false)
```

Comment:
This is a different meaning for `quoteArgs` then we use in the C# extension, and I don't think this is what you want.

This is what I think you want:
  If true: arguments containing spaces get quoted
  If false: arguments are never quoted, even if they have spaces.

Here is the code that I think you want:

```C#
        /// <summary>Enclose the given string in quotes if it contains space or tab characters.</summary>
        public static string Quote(string arg)
        {
            // Check to see if it has already been quoted
            if (arg[0] == '"')
            {
                return arg;
            }

            if (arg.IndexOfAny(ArgumentSeparators) >= 0)
            {
                return '"' + arg + '"';
            }
            return arg;
        }

        /// <summary>Join the arguments list, quoting arguments when necessary.</summary>
        public static string JoinArguments(IEnumerable<string> args, bool withQuote)
        {
            StringBuilder stringBuilder = new StringBuilder();
            if (args != null)
            {
                foreach (string arg in args)
                {
                    if (stringBuilder.Length != 0)
                        stringBuilder.Append(' ');

                    if (withQuote)
                    {
                        stringBuilder.Append(Quote(arg));
                    }
                    else
                    {
                        stringBuilder.Append(arg);
                    }
                }
            }
            return stringBuilder.ToString();
        }
```